pkg/operator/cluster/controller.go

```diff
@@ -37,68 +34,133 @@ import (
 	"github.com/rook/rook/pkg/operator/mgr"
 	"github.com/rook/rook/pkg/operator/mon"
 	"github.com/rook/rook/pkg/operator/osd"
-	"github.com/rook/rook/pkg/operator/rgw"
-	rookclient "github.com/rook/rook/pkg/rook/client"
+	"github.com/rook/rook/pkg/operator/pool"
 	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
 )
 
 const (
-	crushConfigMapName = "crush-config"
-	crushmapCreatedKey = "initialCrushMapCreated"
+	customResourceName       = "cluster"
+	customResourceNamePlural = "clusters"
+	crushConfigMapName       = "crush-config"
+	crushmapCreatedKey       = "initialCrushMapCreated"
+	clusterCreateInterval    = 6 * time.Second
```

Comment:
I dont think we ever had one. The test in https://github.com/rook/rook/blob/master/pkg/operator/operator_test.go just test CustomResource creation. So it is a test for the kit and not for the cluster.